### PR TITLE
Error handling and shared data services

### LINKED ISSUE #2799: Centralize apollo error handling
Noticed that some of the hooks does not handle error handling properly, for example: `useFetchCurrentUser`,  `useFetchGeneralSettings` and `useEvent`(it uses `useCacheQuery`, but this also does not handle errors).
ATM I can see 2 directions:
- add error handling for each of the hooks
- implement an adaptor that will take care of error handling
***
Also we need to handle network errors with smth like `apollo-link-error`:
- https://www.apollographql.com/docs/link/links/error/
- https://github.com/apollographql/apollo-link/tree/master/packages/apollo-link-error

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- **Moves** `/domain/shared/services/apollo` **to** `/data/services/apollo` and fixes paths
- Moves types from `domain/eventEditor/services/apollo` to `data/services/apollo`
- Add error handling to `useFetchCurrentUser`, `useFetchGeneralSettings` and `useEvent`

 Closes #2799

### CODE REVIEW

File: assets/src/data/services/apollo/queries/currentUser/useFetchCurrentUser.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { Viewer } from '@application/valueObjects/config/types';
+
+const useFetchCurrentUser = (): FetchQueryResult<Viewer> => {
+	const toaster = useSystemNotifications();
+
+	const { data, error, loading } = useQuery<Viewer>(GET_CURRENT_USER, {
+		// only display error, not loading or success
+		onError: (error): void => {
+			toaster.error({ message: error.message });
+		},
```
Comment (nicolad) [inline assets/src/data/services/apollo/queries/currentUser/useFetchCurrentUser.ts:15]: :+1:

File: assets/src/data/services/apollo/queries/generalSettings/useFetchGeneralSettings.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { FetchQueryResult } from '@dataServices/apollo/queries/types';
+import { GeneralSettingsData } from '@application/valueObjects/config/types';
+
+const useFetchGeneralSettings = (): FetchQueryResult<GeneralSettingsData> => {
+	const toaster = useSystemNotifications();
+
+	const { data, error, loading } = useQuery<GeneralSettingsData>(GET_GENERAL_SETTINGS, {
+		// only display error, not loading or success
+		onError: (error): void => {
+			toaster.error({ message: error.message });
```
Comment (nicolad) [inline assets/src/data/services/apollo/queries/generalSettings/useFetchGeneralSettings.ts:14]: :+1:

File: assets/src/data/services/apollo/queries/generalSettings/useGeneralSettings.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { GET_GENERAL_SETTINGS } from './';
+import { GET_GENERAL_SETTINGS } from '.';
```
Comment (nicolad) [inline assets/src/data/services/apollo/queries/generalSettings/useGeneralSettings.ts:1]: is this smth OS-related?

Comment (manzoorwanijk) [reply assets/src/data/services/apollo/queries/generalSettings/useGeneralSettings.ts:1]: Something `VS Code` related 🤦‍♂️ 😄 
I let it update the paths

**Pull request merged**